--- a/conda/extensions.bzl
+++ b/conda/extensions.bzl
@@ -9,7 +9,7 @@
         path = "python.exe"
     return rctx.execute(
         [Label("@python_3_11_host//:" + path)] + args,
-        environment = {"PYTHONPATH": str(rctx.path(Label("@rules_conda_pypi_311_py_rattler//:site-packages")))},
+        environment = {"PYTHONPATH": str(rctx.path(Label("@rules_conda_pypi_311_py_rattler_cp38_abi3_manylinux_2_17_x86_64_4729deb0//:site-packages")))},
     )
 
 def _check_result(result, error):
