--- internal/app/machined/pkg/controllers/network/nftables_chain.go	2026-03-04 05:49:49.298666240 +0000
+++ internal/app/machined/pkg/controllers/network/nftables_chain.go	2026-03-04 05:49:49.332666241 +0000
@@ -15,8 +15,6 @@
 	"github.com/google/nftables"
 	"github.com/google/nftables/expr"
 	"go.uber.org/zap"
-
-	networkadapter "github.com/siderolabs/talos/internal/app/machined/pkg/adapters/network"
 	"github.com/siderolabs/talos/pkg/machinery/constants"
 	"github.com/siderolabs/talos/pkg/machinery/resources/network"
 )
@@ -63,11 +61,6 @@
 		}
 
 		var conn nftables.Conn
-
-		if err := ctrl.preCreateIptablesNFTable(logger, &conn); err != nil {
-			return fmt.Errorf("error pre-creating iptables-nft table: %w", err)
-		}
-
 		list, err := safe.ReaderListAll[*network.NfTablesChain](ctx, r)
 		if err != nil {
 			return fmt.Errorf("error listing nftables chains: %w", err)
@@ -120,7 +113,7 @@
 			})
 
 			for _, rule := range chain.TypedSpec().Rules {
-				compiled, err := networkadapter.NfTablesRule(&rule).Compile()
+				compiled, err := NfTablesRule(&rule).Compile()
 				if err != nil {
 					return fmt.Errorf("error compiling nftables rule for chain %s: %w", nfChain.Name, err)
 				}
@@ -177,34 +170,3 @@
 		r.ResetRestartBackoff()
 	}
 }
-
-func (ctrl *NfTablesChainController) preCreateIptablesNFTable(logger *zap.Logger, conn *nftables.Conn) error {
-	// Pre-create the iptables-nft table, if it doesn't exist.
-	// This is required to ensure that the iptables universal binary prefers iptables-nft over
-	// iptables-legacy can be used to manage the nftables rules.
-	tables, err := conn.ListTablesOfFamily(nftables.TableFamilyIPv4)
-	if err != nil {
-		return fmt.Errorf("error listing existing nftables tables: %w", err)
-	}
-
-	if slices.IndexFunc(tables, func(t *nftables.Table) bool { return t.Name == "mangle" }) != -1 {
-		return nil
-	}
-
-	table := &nftables.Table{
-		Family: nftables.TableFamilyIPv4,
-		Name:   "mangle",
-	}
-	conn.AddTable(table)
-
-	chain := &nftables.Chain{
-		Name:  "KUBE-IPTABLES-HINT",
-		Table: table,
-		Type:  nftables.ChainTypeNAT,
-	}
-	conn.AddChain(chain)
-
-	logger.Info("pre-created iptables-nft table 'mangle'/'KUBE-IPTABLES-HINT'")
-
-	return nil
-}
