--- internal/app/machined/pkg/adapters/kubespan/peer_status.go	2026-03-04 05:49:49.015666238 +0000
+++ internal/app/machined/pkg/adapters/kubespan/peer_status.go	2026-03-04 05:49:49.313666240 +0000
@@ -11,8 +11,6 @@
 	"github.com/siderolabs/gen/value"
 	"go4.org/netipx"
 	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
-
-	"github.com/siderolabs/talos/internal/app/machined/pkg/adapters/wireguard"
 	"github.com/siderolabs/talos/pkg/machinery/resources/kubespan"
 )
 
@@ -32,6 +30,15 @@
 // EndpointConnectionTimeout is time to wait for initial handshake when the endpoint is just set.
 const EndpointConnectionTimeout = 15 * time.Second
 
+// PeerDownInterval is the time since last handshake when established peer is considered to be down.
+//
+// WG whitepaper defines a downed peer as being:
+// Handshake Timeout (180s) + Rekey Timeout (5s) + Rekey Attempt Timeout (90s)
+//
+// This interval is applied when the link is already established.
+// Ref: talos/internal/app/machined/pkg/adapters/wireguard/wireguard.go
+const PeerDownInterval = (180 + 5 + 90) * time.Second
+
 // CalculateState updates connection state based on other fields values.
 //
 // Goal: endpoint is ultimately down if we haven't seen handshake for more than peerDownInterval,
@@ -64,9 +71,9 @@
 // CalculateStateWithDurations calculates the state based on the time since events.
 func (a peerStatus) CalculateStateWithDurations(sinceLastHandshake, sinceEndpointChange time.Duration) {
 	switch {
-	case sinceEndpointChange > wireguard.PeerDownInterval: // past T0+peerDownInterval
+	case sinceEndpointChange > PeerDownInterval: // past T0+peerDownInterval
 		// if we got handshake in the last peerDownInterval, endpoint is up
-		if sinceLastHandshake < wireguard.PeerDownInterval {
+		if sinceLastHandshake < PeerDownInterval {
 			a.PeerStatusSpec.State = kubespan.PeerStateUp
 		} else {
 			a.PeerStatusSpec.State = kubespan.PeerStateDown
