diff --git a/proto/BUILD b/proto/BUILD
index 424cdaf..03f374b 100644
--- a/proto/BUILD
+++ b/proto/BUILD
@@ -871,10 +871,6 @@ proto_library(
     srcs = ["search.proto"],
     deps = [
         ":context_proto",
-        "@io_kythe//kythe/proto:common_proto",
-        "@io_kythe//kythe/proto:filetree_proto",
-        "@io_kythe//kythe/proto:graph_proto",
-        "@io_kythe//kythe/proto:xref_proto",
     ],
 )
 
@@ -904,53 +900,10 @@ proto_library(
     ],
 )
 
-ts_proto_library(
-    name = "kythe_common_ts_proto",
-    proto = "@io_kythe//kythe/proto:common_proto",
-    deps = [
-        ":timestamp_ts_proto",
-    ],
-)
-
-ts_proto_library(
-    name = "kythe_graph_ts_proto",
-    proto = "@io_kythe//kythe/proto:graph_proto",
-    deps = [
-        ":kythe_common_ts_proto",
-    ],
-)
-
-ts_proto_library(
-    name = "kythe_xref_ts_proto",
-    proto = "@io_kythe//kythe/proto:xref_proto",
-    deps = [
-        ":kythe_common_ts_proto",
-    ],
-)
-
-ts_proto_library(
-    name = "kythe_identifier_ts_proto",
-    proto = "@io_kythe//kythe/proto:identifier_proto",
-)
-
-ts_proto_library(
-    name = "kythe_filetree_ts_proto",
-    proto = "@io_kythe//kythe/proto:filetree_proto",
-)
-
 proto_library(
     name = "kythe_service_proto",
     srcs = ["kythe_service.proto"],
     deps = [
-        "@io_kythe//kythe/proto:analysis_proto",
-        "@io_kythe//kythe/proto:explore_proto",
-        "@io_kythe//kythe/proto:filetree_proto",
-        "@io_kythe//kythe/proto:graph_proto",
-        "@io_kythe//kythe/proto:identifier_proto",
-        "@io_kythe//kythe/proto:link_proto",
-        "@io_kythe//kythe/proto:status_service_proto",
-        "@io_kythe//kythe/proto:storage_proto",
-        "@io_kythe//kythe/proto:xref_proto",
     ],
 )
 
@@ -1815,10 +1768,6 @@ go_proto_library(
     proto = ":search_proto",
     deps = [
         ":context_go_proto",
-        "@io_kythe//kythe/proto:common_go_proto",
-        "@io_kythe//kythe/proto:filetree_go_proto",
-        "@io_kythe//kythe/proto:graph_go_proto",
-        "@io_kythe//kythe/proto:xref_go_proto",
     ],
 )
 
@@ -1859,15 +1808,6 @@ go_proto_library(
     mode = "services_only",
     proto = ":kythe_service_proto",
     deps = [
-        "@io_kythe//kythe/proto:analysis_go_proto",
-        "@io_kythe//kythe/proto:explore_go_proto",
-        "@io_kythe//kythe/proto:filetree_go_proto",
-        "@io_kythe//kythe/proto:graph_go_proto",
-        "@io_kythe//kythe/proto:identifier_go_proto",
-        "@io_kythe//kythe/proto:link_go_proto",
-        "@io_kythe//kythe/proto:status_service_go_proto",
-        "@io_kythe//kythe/proto:storage_go_proto",
-        "@io_kythe//kythe/proto:xref_go_proto",
     ],
 )
 
