diff --git a/proto/BUILD b/proto/BUILD
index 424cdaf..73d09bd 100644
--- a/proto/BUILD
+++ b/proto/BUILD
@@ -199,7 +199,6 @@ proto_library(
     ],
     deps = [
         ":resource_proto",
-        "@com_github_planetscale_vtprotobuf//include/github.com/planetscale/vtprotobuf/vtproto:vtproto_proto",
     ],
 )
 
@@ -345,7 +344,6 @@ proto_library(
     deps = [
         ":remote_execution_proto",
         ":resource_proto",
-        "@com_github_planetscale_vtprotobuf//include/github.com/planetscale/vtprotobuf/vtproto:vtproto_proto",
     ],
 )
 
@@ -456,7 +454,6 @@ proto_library(
     deps = [
         ":context_proto",
         ":invocation_status_proto",
-        "@com_github_planetscale_vtprotobuf//include/github.com/planetscale/vtprotobuf/vtproto:vtproto_proto",
     ],
 )
 
@@ -836,7 +833,6 @@ proto_library(
     srcs = ["zip.proto"],
     deps = [
         ":context_proto",
-        "@com_github_planetscale_vtprotobuf//include/github.com/planetscale/vtprotobuf/vtproto:vtproto_proto",
     ],
 )
 
@@ -1286,7 +1282,6 @@ go_proto_library(
     deps = [
         ":context_go_proto",
         ":invocation_status_go_proto",
-        "@com_github_planetscale_vtprotobuf//include/github.com/planetscale/vtprotobuf/vtproto:vtproto_go_proto",
     ],
 )
 
@@ -1347,7 +1342,6 @@ go_proto_library(
     deps = [
         ":remote_execution_go_proto",
         ":resource_go_proto",
-        "@com_github_planetscale_vtprotobuf//include/github.com/planetscale/vtprotobuf/vtproto:vtproto_go_proto",
     ],
 )
 
@@ -1661,7 +1655,6 @@ go_proto_library(
     proto = ":distributed_cache_proto",
     deps = [
         ":resource_go_proto",
-        "@com_github_planetscale_vtprotobuf//include/github.com/planetscale/vtprotobuf/vtproto:vtproto_go_proto",
     ],
 )
 
@@ -1775,7 +1768,6 @@ go_proto_library(
     deps = [
         ":context_go_proto",
         ":remote_execution_go_proto",
-        "@com_github_planetscale_vtprotobuf//include/github.com/planetscale/vtprotobuf/vtproto:vtproto_go_proto",
     ],
 )
 
diff --git a/proto/buildbuddy_service.proto b/proto/buildbuddy_service.proto
index 4594942..6dbb418 100644
--- a/proto/buildbuddy_service.proto
+++ b/proto/buildbuddy_service.proto
@@ -280,7 +280,6 @@ service BuildBuddyService {
 
   // Codesearch and Kythe API
   rpc Search(search.SearchRequest) returns (search.SearchResponse);
-  rpc KytheProxy(search.KytheRequest) returns (search.KytheResponse);
   rpc Index(index.IndexRequest) returns (index.IndexResponse);
   rpc RepoStatus(index.RepoStatusRequest) returns (index.RepoStatusResponse);
 
diff --git a/proto/codesearch_service.proto b/proto/codesearch_service.proto
index f7d1d08..21737d1 100644
--- a/proto/codesearch_service.proto
+++ b/proto/codesearch_service.proto
@@ -13,5 +13,4 @@ service CodesearchService {
   rpc Search(search.SearchRequest) returns (search.SearchResponse);
   rpc IngestAnnotations(index.IngestAnnotationsRequest)
       returns (index.IngestAnnotationsResponse);
-  rpc KytheProxy(search.KytheRequest) returns (search.KytheResponse);
 }
diff --git a/proto/distributed_cache.proto b/proto/distributed_cache.proto
index d43decf..51384a1 100644
--- a/proto/distributed_cache.proto
+++ b/proto/distributed_cache.proto
@@ -1,7 +1,6 @@
 syntax = "proto3";
 
 import "proto/resource.proto";
-import "github.com/planetscale/vtprotobuf/vtproto/ext.proto";
 
 package distributed_cache;
 
@@ -18,7 +17,6 @@ message ReadRequest {
 }
 
 message ReadResponse {
-  option (vtproto.mempool) = true;
   bytes data = 1;
 }
 
diff --git a/proto/eventlog.proto b/proto/eventlog.proto
index 5a70bad..b12b0b0 100644
--- a/proto/eventlog.proto
+++ b/proto/eventlog.proto
@@ -2,7 +2,6 @@ syntax = "proto3";
 
 import "proto/context.proto";
 import "proto/invocation_status.proto";
-import "github.com/planetscale/vtprotobuf/vtproto/ext.proto";
 
 package eventlog;
 
@@ -51,7 +50,6 @@ message GetEventLogChunkResponse {
 }
 
 message LiveEventLogChunk {
-  option (vtproto.mempool) = true;
   // The chunk id that this data will eventually be written to.
   // Technically, when written the tail could span to the following chunk, but
   // that is not relevant for our purposes here.
diff --git a/proto/search.proto b/proto/search.proto
index 68190b2..2eec70d 100644
--- a/proto/search.proto
+++ b/proto/search.proto
@@ -1,10 +1,5 @@
 syntax = "proto3";
 
-import "kythe/proto/common.proto";
-import "kythe/proto/filetree.proto";
-import "kythe/proto/graph.proto";
-import "kythe/proto/xref.proto";
-
 import "proto/context.proto";
 
 package search;
@@ -110,67 +105,6 @@ message ExtendedXrefsRequest {
   repeated string tickets = 1;
 }
 
-// See https://kythe.io/docs/schema/ for Kythe schema details.
-message ExtendedXrefsReply {
-  // Definitions of the requested ticket
-  repeated kythe.proto.CrossReferencesReply.RelatedAnchor definitions = 1;
-
-  // Overrides of the requested ticket.
-  // See /kythe/edge/overrides[/transitive]
-  repeated kythe.proto.CrossReferencesReply.RelatedAnchor overrides = 2;
-
-  // Places where the requested ticket is overridden. The inverse of
-  // `overrides`. See /kythe/edge/overrides[/transitive]
-  repeated kythe.proto.CrossReferencesReply.RelatedAnchor overridden_by = 3;
-
-  // References to entities which the requested ticket extends.
-  // See /kythe/edge/extends and /kythe/edge/satisfies.
-  repeated kythe.proto.CrossReferencesReply.RelatedAnchor extends = 4;
-
-  // References to entities which extend the requested ticket. The inverse of
-  // `extends`. See /kythe/edge/extends and /kythe/edge/satisfies.
-  repeated kythe.proto.CrossReferencesReply.RelatedAnchor extended_by = 5;
-
-  // Extended references to the requested ticket. This will include references
-  // to super/sub-classes and overridden/overriding methods.
-  repeated kythe.proto.CrossReferencesReply.RelatedAnchor references = 6;
-  repeated kythe.proto.CrossReferencesReply.RelatedAnchor generated_by = 7;
-}
-
 message ExtendedDocumentationRequest {
   string ticket = 1;
 }
-
-message ExtendedDocumentationReply {
-  string docstring = 1;
-  kythe.proto.common.NodeInfo node_info = 2;
-  kythe.proto.CrossReferencesReply.RelatedAnchor definition = 3;
-}
-
-message KytheRequest {
-  context.RequestContext request_context = 1;
-
-  oneof value {
-    kythe.proto.NodesRequest nodes_request = 2;
-    kythe.proto.DecorationsRequest decorations_request = 3;
-    kythe.proto.CrossReferencesRequest cross_references_request = 4;
-    kythe.proto.CorpusRootsRequest corpus_roots_request = 5;
-    kythe.proto.DirectoryRequest directory_request = 6;
-    ExtendedXrefsRequest extended_xrefs_request = 7;
-    ExtendedDocumentationRequest docs_request = 8;
-  }
-}
-
-message KytheResponse {
-  context.ResponseContext response_context = 1;
-
-  oneof value {
-    kythe.proto.NodesReply nodes_reply = 2;
-    kythe.proto.DecorationsReply decorations_reply = 3;
-    kythe.proto.CrossReferencesReply cross_references_reply = 4;
-    kythe.proto.CorpusRootsReply corpus_roots_reply = 5;
-    kythe.proto.DirectoryReply directory_reply = 6;
-    ExtendedXrefsReply extended_xrefs_reply = 7;
-    ExtendedDocumentationReply docs_reply = 8;
-  }
-}
diff --git a/proto/storage.proto b/proto/storage.proto
index 7b56e57..69b5bcb 100644
--- a/proto/storage.proto
+++ b/proto/storage.proto
@@ -2,7 +2,6 @@ syntax = "proto3";
 
 import "proto/resource.proto";
 import "proto/remote_execution.proto";
-import "github.com/planetscale/vtprotobuf/vtproto/ext.proto";
 
 package storage;
 
@@ -56,7 +55,6 @@ message StorageMetadata {
 }
 
 message FileMetadata {
-  option (vtproto.mempool) = true;
   FileRecord file_record = 1;
   StorageMetadata storage_metadata = 2;
   EncryptionMetadata encryption_metadata = 6;
diff --git a/proto/zip.proto b/proto/zip.proto
index 8babd34..b5eb531 100644
--- a/proto/zip.proto
+++ b/proto/zip.proto
@@ -1,12 +1,10 @@
 syntax = "proto3";
 
 import "proto/context.proto";
-import "github.com/planetscale/vtprotobuf/vtproto/ext.proto";
 
 package zip;
 
 message ManifestEntry {
-  option (vtproto.mempool) = true;
   // The full path of the file as specified in the archive.
   string name = 1;
 
diff --git a/server/interfaces/interfaces.go b/server/interfaces/interfaces.go
index 2bf2370..8d0b65c 100644
--- a/server/interfaces/interfaces.go
+++ b/server/interfaces/interfaces.go
@@ -1811,7 +1811,6 @@ type CodesearchService interface {
 	Index(ctx context.Context, req *csinpb.IndexRequest) (*csinpb.IndexResponse, error)
 	RepoStatus(ctx context.Context, req *csinpb.RepoStatusRequest) (*csinpb.RepoStatusResponse, error)
 	IngestAnnotations(ctx context.Context, req *csinpb.IngestAnnotationsRequest) (*csinpb.IngestAnnotationsResponse, error)
-	KytheProxy(ctx context.Context, req *cssrpb.KytheRequest) (*cssrpb.KytheResponse, error)
 }
 
 type AuthService interface {
